preprocessing/nextclade/src/loculus_preprocessing/processing_functions.py

```diff
@@ -582,16 +587,33 @@ def concatenate(
 
         number_fields = len(input_data.keys()) + 1
 
-        accession_version = args["accession_version"]
+        if not isinstance(args["accession_version"], str):
+            return ProcessingResult(
+                datum=None,
+                warnings=[],
+                errors=[
+                    ProcessingAnnotation(
+                        processedFields=[
+                            AnnotationSource(name=output_field, type=AnnotationSourceType.METADATA)
+                        ],
+                        unprocessedFields=[
+                            AnnotationSource(name=field, type=AnnotationSourceType.METADATA)
+                            for field in input_fields
+                        ],
+                        message=(
+                            f"Internal Error: Function concatenate did not receive accession_version "
+                            f"ProcessingResult with input {input_data} and args {args}, "
+                            "please contact the administrator."
+                        ),
+                    )
+                ],
+            )
+
+        accession_version: str = args["accession_version"]
         order = args["order"]
         type = args["type"]
 
-        # Check accessionVersion only exists once in the list:
-        if number_fields != len(order):
-            logging.error(
-                f"Concatenate: Expected {len(order)} fields, got {number_fields}. "
-                f"This is probably a configuration error. (accession_version: {accession_version})"
-            )
+        def add_errors():
```

Comment:
That's a weird function - rather than calling a magic `add_errors` function, we should just define the error item once, then do `errors.append(standard_error)` - that's a lot more explicit and doesn't use closure